packages/lodestar/src/network/reqresp/rateTracker.ts

```diff
@@ -0,0 +1,48 @@
+import {MapDef} from "../../util/map";
+
+/**
+ * The generic rate tracker allows up to `limit` objects in a period of time.
+ * This could apply to both request count or block count, for both requests and responses.
+ */
+export class RateTracker {
+  private requestsWithinWindow = 0;
+  /** Key as time in ms and value as object requested */
+  private requests = new MapDef<number, number>(() => 0);
+
+  constructor(private limit: number, private timeoutMs: number) {}
+
+  requestObjects(objectCount: number): number {
+    if (objectCount <= 0) throw Error("Invalid objectCount " + objectCount);
+    this.prune();
+    if (this.requestsWithinWindow >= this.limit) {
+      return 0;
+    }
+
+    this.requestsWithinWindow += objectCount;
+    const now = Date.now();
+    const curObjectCount = this.requests.getOrDefault(now);
```

Comment:
`curObjectCount` equals to the count of objects in that exact ms? So we are limiting how many objects can happen in a per milisecond basis?

Reply:
@dapplion 
> curObjectCount equals to the count of objects in that exact ms

yes, we could switch to `second` instead to reduce the memory through

> So we are limiting how many objects can happen in a per milisecond basis?

the quota depends on the `timeoutMs` in the constructor of this class, not the key of the `this.request` map. Again, I think I should switch to tracking by `second` instead of `ms` to make it simpler and take less memory